.github/workflows/pr-check.yml

```
@@ -0,0 +1,46 @@
+
+      - name: Run build
+        run: |
+          pnpm run build
+
+      # Step 2: Check if the branch is up-to-date with preview
+      - name: Fetch preview branch
+        run: git fetch origin preview
+
+      - name: Check if branch is up-to-date with preview
+        id: check_update
+        run: |
```

actionlint

workflow command "set-output" was deprecated. use `echo "{name}={value}" >> $GITHUB_OUTPUT` instead: https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions

(deprecated-commands)